java/core/src/main/java/com/microsoft/bond/BondedBondType.java

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+/**
+ * Implements the {@link BondType} contract for bonded container data type.
+ * @param <TStruct> the class of the underlying struct value
+ */
+public final class BondedBondType<TStruct extends BondSerializable> extends BondType<Bonded<TStruct>> {
+
+    /**
+     * The name of the type as it appears in Bond schemas.
+     */
+    public static final String TYPE_NAME = "bonded";
+
+    private final StructBondType<TStruct> valueType;
+
+    // restrict instantiation to the current package
+    BondedBondType(StructBondType<TStruct> valueType) {
+        super(-valueType.hashCode());
```

Comment:
Obviously you can't have unique hash code, but as distributed as possible for better perf. I'll keep it as it for now and we may revisit hash code computations later as part of perf tuning.

---
In reply to: [123108950](https://github.com/Microsoft/bond/pull/502#discussion_r123108950) [](ancestors = 123108950)